ambry-account/src/main/java/com/github/ambry/account/AccountUtils.java

```diff
@@ -70,4 +73,95 @@ static boolean hasDuplicateAccountIdOrName(Collection<Account> accounts) {
     deprecatedContainers.addAll(accountService.getContainersByStatus(Container.ContainerStatus.INACTIVE));
     return deprecatedContainers;
   }
+
+  /**
+   * Compares and logs differences (if any) in Accounts.
+   * @param accountsInPrimary accounts in primary collection.
+   * @param accountsInSecondary accounts in secondary collection.
+   * @return true if accounts and their containers are same in both collections.
+   */
+  public static boolean compareAccounts(Collection<Account> accountsInPrimary,
+      Collection<Account> accountsInSecondary) {
+    boolean isSame = true;
+    Map<Short, Account> secondaryAccountMap = new HashMap<>();
+    accountsInSecondary.forEach(account -> secondaryAccountMap.put(account.getId(), account));
+
+    Set<Account> accountsMissingInSecondary = accountsInPrimary.stream()
+        .filter(account -> secondaryAccountMap.get(account.getId()) == null)
```

Comment:
Minor: can use !secondaryAccountMap.containsKey(account.getId())

Reply:
Also, we are filtering on accountId here but on containerName below.  Maybe name makes more sense?